### PR TITLE
Mon 4114 update fix for 18.10.x using mysql8

### PR DESCRIPTION
**Pull request by sc979**
# Pull Request Template

## Description

Branch specifically created to adapt upgrade PR to the 18.10.x
`https://github.com/centreon/centreon/pull/7760`
`https://github.com/centreon/centreon/pull/7782`
`https://github.com/centreon/centreon/pull/7780`
**All these PR are already reviewed and approved**

**Fixes** # (mysql 8 upgrade process)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [x] 18.10.x
- [ ] 19.04.x
- [ ] 19.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

- Install a 18.10.0 with a mariaDB database and try to update to the latest version
- Install a 18.10.0 with a deported mysql database and try to update to the latest version
The scripts should not fail at all.

#### Community contributors & Centreon team

- [x] I followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: cron/centAcl.php [hunk 1]
```diff
@@ -112,15 +112,15 @@ function programExit($msg)
...
+            "UPDATE cron_operation SET `running` = '1', `time_launch` = '" . time() .
             "' WHERE id = '$appID'"
         );
     } else {
         if ($nbProc <= 1) {
             $errorMessage = "According to DB another instance of centAcl.php is already running and I found " .
                 $nbProc . " process...\n";
-            $errorMessage .= "Executing query: UPDATE cron_operation SET running = 0 WHERE id =  '$appID'";
-            $pearDB->query("UPDATE cron_operation SET running = '0' WHERE id = '$appID'");
+            $errorMessage .= "Executing query: UPDATE cron_operation SET `running` = 0 WHERE id =  ' . $appID . '";
```
Comment (loiclau) [inline cron/centAcl.php]: dont add dot

File: cron/centAcl.php [hunk 2]
```diff
@@ -112,15 +112,15 @@ function programExit($msg)
...
+            $pearDB->query("UPDATE cron_operation SET running = '0' WHERE id = ' . $appID . '");
```
Comment (loiclau) [inline cron/centAcl.php]: dont add dot

File: www/class/centreon-partition/partEngine.class.php [hunk 3]
```diff
@@ -606,18 +606,50 @@ public function backupParts($table, $db)
...
+            $dbType = $dbVersion = null;
+            while ($row = $dbResult->fetch()) {
+                switch ($row['Variable_name']) {
+                    case 'version_comment' :
+                        $dbType = $row['Value'];
+                        break;
+                    case 'version' :
+                        $dbVersion = $row['Value'];
+                        break;
+                    default :
```
Comment (loiclau) [inline www/class/centreon-partition/partEngine.class.php]: in what case you can go in default?

File: www/class/centreon-partition/partEngine.class.php [hunk 4]
```diff
@@ -606,18 +606,50 @@ public function backupParts($table, $db)
...
-        if ($config["plugin_status"] != "ACTIVE") {
-            return (false);
+        $dbResult = $db->query("SELECT plugin_status FROM INFORMATION_SCHEMA.PLUGINS WHERE plugin_name = 'partition'");
+        $config = $dbResult->fetch();
+        $dbResult->closeCursor();
+        if ($config["plugin_status"] == "ACTIVE") {
+            unset($config);
+
+            return true;
         }
```
Comment (loiclau) [inline www/class/centreon-partition/partEngine.class.php]: line

File: cron/centAcl.php (hunk 2 above)
Comment (sc979) [inline cron/centAcl.php]:
```suggestion
            $pearDB->query("UPDATE cron_operation SET running = '0' WHERE id = '$appID'");
```

File: cron/centAcl.php (hunk 1 above)
Comment (sc979) [inline cron/centAcl.php]:
```suggestion
            $errorMessage .= "Executing query: UPDATE cron_operation SET `running` = 0 WHERE id =  '$appID'";
```

File: www/class/centreon-partition/partEngine.class.php (hunk 3 above)
Comment (sc979) [inline www/class/centreon-partition/partEngine.class.php]:
```suggestion
```

File: www/class/centreon-partition/partEngine.class.php [hunk 5]
```diff
@@ -606,18 +606,50 @@ public function backupParts($table, $db)
...
+            while ($row = $dbResult->fetch()) {
+                switch ($row['Variable_name']) {
+                    case 'version_comment' :
+                        $dbType = $row['Value'];
+                        break;
+                    case 'version' :
+                        $dbVersion = $row['Value'];
+                        break;
+                    default :
+                        break;
```
Comment (sc979) [inline www/class/centreon-partition/partEngine.class.php]:
```suggestion
```

File: www/class/centreon-partition/partEngine.class.php [hunk 6]
```diff
@@ -606,18 +606,50 @@ public function backupParts($table, $db)
...
+                }
+            }
+            $dbResult->closeCursor();
 
-        return (true);
+            if (stristr($dbType, "MySQL")
+                && (floatval($dbVersion) > 5.9)
+            ) {
+                unset($config, $row);
+
```
Comment (sc979) [inline www/class/centreon-partition/partEngine.class.php]:
```suggestion
```

File: www/class/centreon-partition/partEngine.class.php [hunk 7]
```diff
@@ -606,18 +606,50 @@ public function backupParts($table, $db)
...
-        $config = $DBRESULT->fetchRow();
-        $DBRESULT->closeCursor();
-        if ($config["plugin_status"] != "ACTIVE") {
-            return (false);
+        $dbResult = $db->query("SELECT plugin_status FROM INFORMATION_SCHEMA.PLUGINS WHERE plugin_name = 'partition'");
+        $config = $dbResult->fetch();
+        $dbResult->closeCursor();
+        if ($config["plugin_status"] == "ACTIVE") {
+            unset($config);
+
```
Comment (sc979) [inline www/class/centreon-partition/partEngine.class.php]:
```suggestion
```

File: www/class/centreon-partition/partEngine.class.php (hunk 4 above)
Comment (sc979) [reply www/class/centreon-partition/partEngine.class.php]: are you talking about the line between the unset and the return ? If so, I removed it ;)

Comment (sc979) [inline www/class/centreon-partition/partEngine.class.php]:
```suggestion
```

File: www/class/centreon-partition/partEngine.class.php [hunk 8]
```diff
@@ -606,18 +606,50 @@ public function backupParts($table, $db)
...
+            return true;
         }
-        unset($config);
+        elseif (empty($config["plugin_status"])) {
```
Comment (sc979) [inline www/class/centreon-partition/partEngine.class.php]:
```suggestion
        } elseif (empty($config["plugin_status"])) {
```

File: cron/centAcl.php [hunk 9]
```diff
@@ -112,15 +112,15 @@ function programExit($msg)
 
     if ($is_running == 0) {
         $DBRESULT = $pearDB->query(
-            "UPDATE cron_operation SET running = '1', time_launch = '" . time() .
+            "UPDATE cron_operation SET `running` = '1', `time_launch` = '" . time() .
             "' WHERE id = '$appID'"
```
Comment (lgcosta) [inline cron/centAcl.php]: Is `$appID` a string from Centreon env or a variable from PHP ? If it is a PHP variable, I think it is more "visual" to separate from the query string, concatenating the variable. (correct form :smile: )

Comment (sc979) [reply cron/centAcl.php]: Some of "system", "running" and "activate" field's names seems to be key words in mysql.
So I added back quote on them...

**Pull request merged**